### PR TITLE
Community Solutions Tag Filter Component

### PR DESCRIPTION
**Pull request by dem4ron**
https://github.com/exercism/website/assets/66035744/0a0341b7-448a-4746-b334-0a3108a9664c

### CODE REVIEW

File: app/javascript/components/track/ExerciseCommunitySolutionsList.tsx [hunk 1]
```diff
@@ -32,8 +33,10 @@ const DEFAULT_ORDER = 'most_starred'
 
 export function ExerciseCommunitySolutionsList({
   request: initialRequest,
+  tags,
 }: {
   request: Request
+  tags: any
```
Comment (dem4ron) [inline app/javascript/components/track/ExerciseCommunitySolutionsList.tsx:43]:
```suggestion
  tags: Tags
```

File: app/javascript/components/track/exercise-community-solutions-list/exercise-tag-filter/ExerciseTagFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import { usePanel } from '@/hooks/use-panel'
+import { Icon } from '@/components/common'
+import { Tags } from './ExerciseTagFilter.types'
+import { ExerciseTagFilterGroup } from './ExerciseTagFilterGroup'
+import { useExerciseTagFilter } from './useExerciseTagFilter'
+import { Request } from '@/hooks/request-query'
```
Comment (dem4ron) [inline app/javascript/components/track/exercise-community-solutions-list/exercise-tag-filter/ExerciseTagFilter.tsx:7]:
```suggestion
import type { Request } from '@/hooks/request-query'
```

Comment (dem4ron) [inline app/javascript/components/track/exercise-community-solutions-list/exercise-tag-filter/ExerciseTagFilter.tsx]:
```suggestion
import React from 'react'
import { usePanel } from '@/hooks/use-panel'
import { Icon } from '@/components/common'
import { ExerciseTagFilterGroup } from './ExerciseTagFilterGroup'
import { useExerciseTagFilter } from './useExerciseTagFilter'
import type { Request } from '@/hooks/request-query'
import type { Tags } from './ExerciseTagFilter.types'
```

File: app/javascript/components/track/exercise-community-solutions-list/exercise-tag-filter/ExerciseTagFilterGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      {tagCategory[groupName].map((tagName, index) => (
+        <Checkbox
+          key={`${tagName}_${index}`}
+          checked={tagState.indexOf(tagName) !== -1}
+          setChecked={(isChecked) => {
+            handleToggleTag(tagName, isChecked)
+          }}
+          className="leading-160"
+        >
+          {tagName.split(':')[1]}
```
Comment (dem4ron) [inline app/javascript/components/track/exercise-community-solutions-list/exercise-tag-filter/ExerciseTagFilterGroup.tsx]: Maybe turn it to title-case?

Comment (iHiD) [reply app/javascript/components/track/exercise-community-solutions-list/exercise-tag-filter/ExerciseTagFilterGroup.tsx]: Yes please.

**Pull request merged**